.github/workflows/timezone-update.yml

```diff
@@ -0,0 +1,39 @@
+name: Timezone update
+on:
+  schedule:
+    # Run once a week at 00:05 AM UTC on Sunday.
+    - cron: 5 0 * * 0
```

Comment:
these releases are so infrequent, max 5 times a year. Probably better to just run on workflow dispatch. 

If you do want to check weekly, it might be better to use caching and cache the icu-data checkout, and then do an update.  Maybe there's some other way to trigger this.

Reply:
Thanks, @srl295. I'm a bit confused about the caching. Every time we run this we want to ensure that we are fetching the latest data from ICU upstream & then find if we have something new in the repo. 
I'll just change it to workflow dispatch/manually trigerred.

Reply:
If it's manually, we might miss the updates & only know when someone reports that a new version is out. The goal of this PR was to have the workflow tell us that a new version is out.

Reply:
I think having it run weekly is OK if we don't have any other way to trigger it when there's actually an update.

Reply:
What I meant is if the repository was cached it wouldn't have to redownload every time. 

@yumaoka are updates always mentioned on icu-announce?  If so then subscribers to that wouldn't miss it.